main/acle.rst

```diff
@@ -404,6 +408,129 @@ Scalable Vector Extensions (SVE)
 ACLE support for SVE is defined in the Arm C Language Extensions for SVE
 document [SVE-ACLE]_ available on the Arm Developer Website.
 
+.. _ssec-NEONSVEBridge:
+
+NEON-SVE Bridge
+---------------
+
+The NEON_SVE Bridge adds intrinsics that allow conversions between NEON and
+SVE vectors.
+
+These intrinsics are defined in the header file ``arm_extensions_bridge.h``.
```

Comment:
Perhaps this should be called something like arm_sve_neon.h or arm_neon_sve.h instead.  I think it will/should include both arm_neon.h and arm_sve.h.

The problem with a general header file like arm_extensions_bridge.h is that, over time, the header might accrue other unrelated features that the user wasn't expecting, possibly “infecting” the user's namespace with unforeseen styles of identifier.  For xample, suppose “arm_extensions_bridge.h” already existed for an earlier feature and that we then added the Neon/SVE bridge to it.  Users of the original version who upgrade their toolchain might find that arm_neon.h and arm_sve.h are now being included unexpectedly.

Reply:
> Perhaps this should be called something like arm_sve_neon.h or arm_neon_sve.h instead. I think it will/should include both arm_neon.h and arm_sve.h.
If it does include those headers internally, will this not enlarge the total bytes processed by the compiler a fair amount? In total for clang this is 4MiB of source. If this header includes it, and the user includes it, then the compiler is going to have to pull in 8MiB of source. Even if this is not fully parsed it does represent a fair amount of extra work for every compilation unit which might include this header. I don't know for sure if this is worth being concerned about, but it seems we can make that problem go away if the new header forward declares the types it needs internally. Do you have any opinions on this problem and proposed solution?

Reply:
I think it's fair to assume that people will only use the Neon/SVE bridge if they are also using Neon and SVE individually.  So from a user-interface perspective, I think we should document in the ACLE that including the new header file also has the effect of including arm_neon.h and arm_sve.h (just like including arm_sve.h is defined to have the effect of including arm_bf16.h when bfloat16 is supported).  Users can then write:

```
#include <new_header.h>
```

rather than having to write:

```
#include <arm_neon.h>
#include <arm_sve.h>
#include <new_header.h>
```

Even if they do (redundantly) include all three, I think the preprocessor has an optimisation that avoids reading a file a second time if all non-writespace/comment characters were protected by a macro guard.  (At least, GCC does this.  I assume Clang does too. :-))

Reply:
👍 thanks for the input, SGTM.

Reply:
I've changed the header file's name from `arm_extensions_bridge.h` to `arm_neon_sve_bridge.h`. I've updated the spec to say it now includes `arm_neon.h` and `arm_sve.h`